src/cursor/cur_file.c

```diff
@@ -15,6 +15,77 @@
 WT_STAT_USECS_HIST_INCR_FUNC(opread, perf_hist_opread_latency, 100)
 WT_STAT_USECS_HIST_INCR_FUNC(opwrite, perf_hist_opwrite_latency, 100)
 
+/*
+ * Wrapper for substituting checkpoint state when doing checkpoint cursor operations.
+ *
+ * A checkpoint cursor has up to three extra things in it: a dummy transaction (always), a read
+ * timestamp (sometimes), and a dhandle for the corresponding history store checkpoint (sometimes).
+ * If there's a transaction, we substitute the transaction into the session, the read timestamp into
+ * txn_shared (if it's present), and stick the checkpoint name of the history store dhandle in the
+ * session for when the history store is opened, then undo it all after the operation completes.
+ *
+ * If the current transaction is _already_ a checkpoint cursor dummy transaction, however, do
+ * nothing. This happens when the history store logic opens history store cursors inside checkpoint
+ * cursor operations on the data store, and we want to keep the existing state.
+ *
+ * Note: it's important that the anachronistic flag be ordered with respect to the read timestamp if
```

Comment:
Something that was removed :-)  That section of the comment is outdated and was removed in 96bbabd.

(It used to be that the checkpoint's read timestamp got into txn_shared, which then required logic to avoid messing up the pinned/oldest computation. That all got dumped.)